SpiceSharp/Components/Semiconductors/MOS/Level1/TemperatureBehavior.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using SpiceSharp.Diagnostics;
+using SpiceSharp.Attributes;
+using SpiceSharp.Simulations;
+using SpiceSharp.Behaviors;
+
+namespace SpiceSharp.Components.MosfetBehaviors.Level1
+{
+    /// <summary>
+    /// Temperature behavior for a <see cref="Mosfet1"/>
+    /// </summary>
+    public class TemperatureBehavior : Behaviors.TemperatureBehavior
+    {
+        /// <summary>
+        /// Necessary behaviors
+        /// </summary>
+        BaseParameters bp;
+        ModelBaseParameters mbp;
+        ModelTemperatureBehavior modeltemp;
+
+        /// <summary>
+        /// Shared variables
+        /// </summary>
+        [PropertyName("sourcevcrit"), PropertyInfo("Critical source voltage")]
+        public double SourceVCritical { get; protected set; }
```

Comment:
@svenboulanger Why setters are protected?

Reply:
I wanted to give the option to override these properties for customized models.